PodcastGenerator/core/episodes.php

```diff
@@ -92,6 +92,106 @@ function getEpisodes($category = null, $_config)
     return $episodes_data;
 }
 
+function searchEpisodes($name = "", $_config)
+{
+    $name = strtolower($name);
+    $supported_extensions = simplexml_load_file($_config['absoluteurl'] . 'components/supported_media/supported_media.xml');
+    $realsupported_extensions = array();
+    foreach ($supported_extensions as $item) {
+        array_push($realsupported_extensions, $item->extension);
+    }
+    $supported_extensions = $realsupported_extensions;
+    unset($realsupported_extensions);
+
+    // Get episodes names and pubDates (which are the file
+    // modification times).  We'll ignore files with future
+    // timestamps.
+    $now_time = time();
+    $episodes_mtimes = array();
+    if ($handle = opendir($_config['absoluteurl'] . $_config['upload_dir'])) {
+        while (false !== ($entry = readdir($handle))) {
+            // If the file is a 'real' file, has a linked XML file,
+            // and isn't from the future, add its name and
+            // modification time to our array.
+            $this_entry = $_config['absoluteurl'] . $_config['upload_dir'] . $entry;
+            $this_mtime = filemtime($this_entry);
+            if (in_array(pathinfo($this_entry, PATHINFO_EXTENSION), $supported_extensions)
+                && file_exists($_config['absoluteurl'] . $_config['upload_dir'] . pathinfo($this_entry, PATHINFO_FILENAME) . '.xml')
+                && $this_mtime <= $now_time) {
```

Comment:
@emilengler @kalvinmattheus what is the drawback to including future timestamps and do we think that including that might be beneficial to the end-user?

Reply:
There is no drawback, it's just that it is the outscope of this PR. Too much code relies on the old way where episodes in the future are ignored until that date is reached. However I am open to potential fixes. Maybe list episodes in the future if the user is signed in with an admin account?

Reply:
Signed in admin accounts will now see future timestamped episodes as well from [commit aed82f0](https://github.com/PodcastGenerator/PodcastGenerator/pull/290/commits/aed82f0171a2db37dd3eda8967f0e312957f432c) and [commit 1931cc2](https://github.com/PodcastGenerator/PodcastGenerator/pull/290/commits/1931cc2fdfa836d22c6560aa9873f6dee45ba689).